src/tests/routes/crates/list.rs

```diff
@@ -791,12 +791,13 @@ fn seek_based_pagination() {
         if let Some(new_url) = resp.meta.next_page {
             assert_that!(resp.crates, len(eq(1)));
             url = Some(new_url);
+            assert_eq!(resp.meta.total, 3);
         } else {
             assert_that!(resp.crates, empty());
+            assert_eq!(resp.meta.total, 0);
```

Comment:
I don't think I understand this assertion change. shouldn't `total` still show `3`?

Reply:
Previously, the `total` was retrieved through separated db request. After the refactoring, we use single query to fetch data and `total`, but it now includes empty rows. This caused the `total` value becomes 0 and mirrors the behavior of offset-based pagination.